project/MimaExcludes.scala

```diff
@@ -1747,6 +1747,17 @@ object MimaExcludes {
       ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("org.apache.spark.ml.param.shared.HasLoss.org$apache$spark$ml$param$shared$HasLoss$_setter_$loss_="),
       ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("org.apache.spark.ml.param.shared.HasLoss.getLoss"),
       ProblemFilters.exclude[InheritedNewAbstractMethodProblem]("org.apache.spark.ml.param.shared.HasLoss.loss")
+    ) ++ Seq(
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.DerbyDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.TeradataDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.OracleDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.PostgresDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.DB2Dialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.MySQLDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.MsSqlServerDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.JdbcDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.NoopDialect.getCatalystType"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.sql.jdbc.Dialect.getCatalystType")
```

Comment:
This change breaks API compatibility for `JdbcDialect` and its subclasses but they are developer API.
So I'd like to discuss whether it's acceptable to do it or not.

Reply:
Please add JIRA ID there at the beginning of this block. BTW, why do we need to add this `2.0.x` exclusion rule?
```
  // Exclude rules for 2.0.x
  lazy val v20excludes = {
```

Reply:
Thanks, it's just a mistake. I'll fix it.